compiler/x/codegen/BinaryCommutativeAnalyser.cpp

```diff
@@ -181,11 +192,19 @@ void TR_X86BinaryCommutativeAnalyser::genericAnalyser(TR::Node       *root,
          {
          generateRegMemInstruction(regMemOpCode, root, secondRegister, tempMR, _cg);
          }
-      root->setRegister(secondRegister);
+      targetRegister = secondRegister;
       tempMR->decNodeReferenceCounts(_cg);
       notReversedOperands();
       }
 
+   if (!root->getOpCode().isArithmetic())
```

Comment:
yea, I agree. Providing 2 wrapper functions would make the API much cleaner. Thanks!